check_file.py

```diff
@@ -9,29 +9,30 @@
 # Description	: Check a file exists and that we can read the file
 
 import sys		# Import the Modules
-import os		# Import the Modules
-
-# Readfile Functions which open the file that is passed to the script
-
-def readfile(filename):
-	f = open(filename, 'r')
-	line = f.read()
-	print line
-
-def main():
-  if len(sys.argv) == 2:		# Check the arguments passed to the script
-    filename = sys.argv[1]		# The filename is the first argument
-    if not os.path.isfile(filename):	# Check the File exists
-      print '[-] ' + filename + ' does not exist.'
-      exit(0)
-    if not os.access(filename, os.R_OK):	# Check you can read the file
-      print '[-] ' + filename + ' access denied'
-      exit(0)
-  else:
-    print '[-] Usage: ' + str(sys.argv[0]) + ' <filename>' # Print usage if not all parameters passed/Checked
-    exit(0)
-  print '[+] Reading from : ' + filename	# Display Message and read the file contents
-  readfile(filename)
-  
-if __name__ == '__main__':
-  main()
\ No newline at end of file
+import os		# Import the Modules
+
+# Readfile Functions which open the file that is passed to the script
+
+def readfile(filename):
+	try:
+		f = open(filename, 'r')
+		line = f.read()
+		print line
+	except IOError:
+		print 'Error opening the file'

```

Comment:
Here you forget to close the file after "read" it.Could we add a "finally" syntax ?And what if I read a file which is up to several G and may make it out of memory ?